### DISCUSSION TITLE
Video inference support is live :rocket:

### DISCUSSION THREAD

**Discussion by fcakyon** [obss/sahi#626] [Announcements]
Thanks to @madenburak, starting from v0.9.3, `sahi` brings full support for sliced/standard inference for video input format for all MMDetection/Detectron2/YOLOv5 models :fire: 

![sahi-video-inf](https://user-images.githubusercontent.com/34196005/167277618-9552add8-4f7f-49b9-ada4-1047ca3ce754.gif)

- Just give video file as source:

```bash
sahi predict --model_path yolov5s.pt --model_type yolov5 --source video.mp4
``` 

You can also view video render during video inference with `--view_video`:

```bash
sahi predict --model_path yolov5s.pt --model_type yolov5 --source video.mp4 --view_video
``` 

- To `forward 100 frames`, on opened window press key `D `
- To `revert 100 frames`, on opened window press key `A`
- To `forward 20 frames`, on opened window press key `G`
- To `revert 20 frames`, on opened window press key `F`
- To `exit`, on opened window press key `Esc`

Note: If `--view_video` is slow, you can add `--frame_skip_interval=20` argument to skip interval of 20 frames each time.

**Comment by kiki4488** [comment]
Hello, we were trying to do object detection task with video file. But we followed your command but it didn't work well. For example, we put a video file, new video was created. we checked that file, not have detected box.

When we used yolov5's video detected module, that did work well...

How to use yolov5+Sahi for video ?

**Comment by fcakyon** [comment]
@kiki4488 can you write your parameters and code used for inference?

**Comment by kiki4488** [comment]
@fcakyon 
# set detection model
detection_model = Yolov5DetectionModel(
    config_path='../config/coco.yaml',
    model_path='../model/best.pt',
    confidence_threshold=0.3,
    device="cuda:0", # or 'cuda:0'
)

# prediction for video
predict(
    detection_model = detection_model,
    source='../data/test/test_2.mp4',
    slice_height=256,
    slice_width=256,
    overlap_height_ratio=0.2,
    overlap_width_ratio=0.2,
    return_dict=True,
    view_video=False,
    frame_skip_interval=20
)

**Comment by fcakyon** [comment]
@kiki4488 what is your video size? Can you send the link for video file so that I can test it?

**Comment by kiki4488** [comment]
@fcakyon our video's width is 528 , and height is 888 
Is there a problem with the size?
The video is copyrighted, I'm so sorry..

**Comment by Akansel-AKYOL** [comment]
Is it possible to get video inference with class filtering? Normally yolov5 provides this by using --classes argument in detect.py file. How can I this with sahi's predict function. for example the model detects 80 classes but I want to use 5 classes

**Reply by fcakyon** [reply]
@Akansel-AKYOL for yolov5 models you can do it as:

```python
from sahi import AutoDetectionModel
from sahi.predict import predict
import yolov5

model = yolov5.load('yolov5s.pt')

model.conf = 0.25  # NMS confidence threshold
model.iou = 0.45  # NMS IoU threshold
model.classes = [0, 1, 5]  # perform detection on only several classes

detection_model = AutoDetectionModel.from_pretrained(
    model_type='yolov5',
    model=model,
)

predict(
    detection_model=detection_model,
    other parameters ...
)
```

**Reply by Akansel-AKYOL** [reply]
@fcakyon thank you

**Comment by abhavgoel** [comment]
@fcakyon how can i modify confidence_threshold for video inference? i tried changing it in predict.py, ut it doesnt seem to work

**Comment by PankajBarai** [comment]
Can we used this for real time..webcam?

**Reply by AhmedIbrahimai** [reply]
yes , you can use this code 
https://github.com/AhmedIbrahimai/SAHI-and-YOLO8-in-real-time

**Comment by anuragrawal2024** [comment]
@fcakyon Can you please look into #944 and tell me what you think? Thanks!

**Comment by AhmedIbrahimai** [comment]
sahi with yolov8 in real time 
https://github.com/AhmedIbrahimai/SAHI-and-YOLO8-in-real-time
